### ISSUE TITLE
TopLevelPropertyNaming ease UPPER_CASE notation on vals

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#769]
Sometimes I want to have constants which Kotlin can't turn into constants. Such as instantiations of classes. `private val WHITESPACE_REGEX = Regex("\\s")`

What do you think about easing the `TopLevelPropertyNaming` rule here since in traditional Java this would be a final const.

**Comment by arturbosch** [comment]
We're changed this to be intellij compliant. So if intellij changes its inspection, we should do it too.
So I'm against your proposal :/

**Comment by vanniktech** [comment]
Alright works for me.

**Comment by tasomaniac** [comment]
Are you sure IntelliJ does that? When I have top level val which is not const upper case seems to be fine. Also in IntelliJ naming settings, there is no special case option for top level declarations at all. 

This works 

```
val FOO = Foo()
```

**Comment by arturbosch** [comment]
And it also works with detekt? Could you elaborate on your question?

**Comment by sschuberth** [comment]
I actually have exactly the same use case as @vanniktech: detekt was complaining about https://github.com/heremaps/oss-review-toolkit/blob/ff7467f8677f224e43fcd9388d51048b5fe4d6a9/test-utils/src/main/kotlin/Utils.kt#L33-L40 while IntelliJ was not, and when digging into detekt's source code I figured this is only due to "_" not being allowed as a non-leading character in private top-level properties, which clearly looks like an oversight to me and is addressed in my PR #1556.

**Comment by tasomaniac** [comment]
From the original issue above I got the impression that upper case is not allowed on detekt on top-level properties. If they're allowed, its all fine then. IntelliJ allows them.

**Comment by sschuberth** [comment]
Yes, detekt also allows them.
